dotNETStandard/HealthVault.Mobile/cs/Xamarin.Core/ViewModels/ActionPlanDetailsViewModel.cs

```diff
@@ -38,11 +38,9 @@ public ActionPlanDetailsViewModel(ActionPlanInstance actionPlanInstance, IHealth
                 var xpath = task.TrackingPolicy.TargetEvents.FirstOrDefault().ElementXPath;
                 if (xpath.Contains("thing/data-xml/weight"))
                 {
-                    PersonInfo personInfo = await this.connection.GetPersonInfoAsync();
-
                     var weightAddPage = new WeightAddPage
                     {
-                        BindingContext = new WeightAddViewModel(this.connection.CreateThingClient(), personInfo.SelectedRecord.Id, this.NavigationService),
+                        BindingContext = new WeightAddViewModel(this.connection, this.NavigationService),
```

Comment:
>this.connection [](start = 64, length = 15)

injecting bloated objects is not a good pattern.  IMO It's better to inject few more objects but reduce the dependency graph. Here's a good video on the subject [https://www.youtube.com/watch?v=acjvKJiOvXw](https://www.youtube.com/watch?v=acjvKJiOvXw)